core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/ColumnPageStatsVO.java

```diff
@@ -96,8 +96,7 @@ public void update(Object value) {
       case DOUBLE:
         max = ((double) max > (double) value) ? max : value;
         min = ((double) min < (double) value) ? min : value;
-        int num = getDecimalCount((double) value);
-        decimal = decimal > num ? decimal : num;
+        decimal = getDecimalCount((double) value);
```

Comment:
for the following value 775678765456789098765432.789 the num variable would be -8 and according to the earlier logic it would update the decimal value to 0

decimal = 0 > -8 ? 0 : -8
thus the double value would be converted to another dataype that we dont want.